base/server/tomcat/src/com/netscape/cms/tomcat/NuxwdogPasswordStore.java

```diff
@@ -73,17 +80,20 @@ private void addTag(String tag) {
         }
     }
 
+    @Override
     public String getPassword(String tag, int iteration) {
         if (pwCache.containsKey(tag)) {
             return pwCache.get(tag);
         }
 
-        String prompt = PROMPT_PREFIX + tag + ":";
-        if (StringUtils.isNotEmpty(instanceId)) {
-            prompt = "[" + instanceId + "] " + prompt;
-        }
+        String pwd = null;
 
-        String pwd = WatchdogClient.getPassword(prompt, iteration);
+        try {
+            String keyringTag = instanceId + "/" + tag;
+            pwd = Keyring.getPassword(keyringTag, "");
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
```

Comment:
This try-catch block will catch and discard the exception, then the program will continue as if there was no error. Since IllegalArgumentException is a RuntimeException, I'd suggest removing the try-catch block and let the exception be handled by the caller.